File: Car Simulation/sphere.py
from util import getDistanceFromCoordinate
import numpy as np
class Sphere:
    def __init__(self, radius, coords):
        self.radius = radius
        self.coords = np.array([coords[0], coords[1]])

    def get_distance(self, x, y) -> float:
        return getDistanceFromCoordinate(self.coords[0], self.coords[1], x, y)

    def get_distance_coords(self, coords) -> float:
        return getDistanceFromCoordinate(self.coords[0], self.coords[1], coords[0], coords[1])
    
    def isTouching(self, Sphere) -> bool:
        return self.get_distance(Sphere.coords[0], Sphere.coords[1]) <= self.radius + Sphere.radius

    def isInside(self, Sphere) -> bool:
        return self.get_distance(Sphere.coords[0], Sphere.coords[1]) <= Sphere.radius

    def isOutside(self, Sphere) -> bool:
        return self.get_distance(Sphere.coords[0], Sphere.coords[1]) >= Sphere.radius
    

File: BasicOperationPlus/src/Simulation/Sphere/Sphere.cpp
#include "Sphere.h"
#include "../../Util/EuclidianDistance/Distance.h"

Sphere::Sphere(uint8_t radius, uint16_t xcoord, uint16_t ycoord){
    this->radius = radius;
    this->xcoord = xcoord;
    this->ycoord = ycoord;

};

Sphere::~Sphere(){};

void Sphere::setCoords(uint16_t x, uint16_t y){
    this->xcoord = x;
    this->ycoord = y;
};

std::tuple<uint16_t, uint16_t> Sphere::getCoords(void){
    return std::make_tuple(this->xcoord, this->ycoord);
};

double Sphere::getDistance(uint16_t x, uint16_t y){
    return getDistanceFromCoordinate(this->xcoord, this->ycoord, x, y);
};

double Sphere::getDistance(std::tuple<uint16_t, uint16_t> coords){
    return this->getDistance(std::get<0>(coords), std::get<1>(coords));
};

uint8_t Sphere::getRadius(){
    return this->radius;
};

bool Sphere::isTouching(Sphere otherSphere){
    return this->getDistance(otherSphere.getCoords()) <= this->radius + otherSphere.getRadius();
};

bool Sphere::isInside(Sphere otherSphere){
    return this->getDistance(otherSphere.getCoords()) <= otherSphere.getRadius();
};

bool Sphere::isOutside(Sphere otherSphere){
    return this->getDistance(otherSphere.getCoords()) >= otherSphere.getRadius();
};
